keras_nlp/samplers/beam_sampler.py

```diff
@@ -85,7 +85,13 @@ def get_next_token(self, next_token_probs):
         pass
 
     def sample(
-        self, prompt, token_probability_fn, mask, num_steps, from_logits=True
+        self,
```

Comment:
Let's just call this `return_all_beams`.

I think rather than updating the `sample()` API, you should add `return_all_beams` as an init arg, and add `self.return_all_beams` to the config.

I also don't think this will work quite yet. The sampler API will generally be called via `__call__`, which will expect only a single output from sample. I will think more about how we can best handle this.

Reply:
Hey!
Sure, I shall make the changes. Should I make them regardless of the sampler API call or will we have to leave this issue to be resolved later?

Reply:
I think we may need to pause on this for a second, though I will draft a PR shortly that should allow us to have more dynamic return types from sampler layers. Will keep you posted!

Reply:
I think this should be quite doable after https://github.com/keras-team/keras-nlp/pull/804 lands.

Reply:
@mattdangerw Since #804 has landed, should I work on this issue further now? Are there any new things you would like me to take care of and handle? Thanks!

Reply:
Yeah! Let's do it.

Not that beam search just overrides call, it should be very easy to return differently based on a boolean parameter.

Let's take in `return_all_beams` in `__init__`, add it to the config for the model, and use that parameter to control the return type during `__call__`. 

We should also make sure to add unit test for this. Let us know when this is ready for review!